libs/admin/src/lib/components/users/users.component.ts

```diff
@@ -36,24 +58,84 @@ export class UsersComponent implements OnInit, OnDestroy {
     this.initSubscriptions();
   }
 
+  public ngAfterViewInit() {
+    this.dataSource.paginator = this.paginator;
+  }
+
   public ngOnDestroy(): void {
     this.subscriptions.unsubscribe();
   }
 
-  public *getUserWorkgroups(user: User): Iterable<Workgroup & { role: Role }> {
-    const iter = user.roles.entries();
-    for (let i = 0; i < this.WORKGROUP_DISPLAY_COUNT; i++) {
-      const { value, done } = iter.next();
-      if (done) {
-        break;
+  public setSearchTerm(term: string) {
+    this.searchTerm$.next(term);
+  }
+
+  public toggleFilters() {
+    this.showFilters = !this.showFilters;
+    if (!this.showFilters) {
+      this.activeFilters$.next(new Map());
+    }
+  }
+
+  public setFilters(selectedValues: Filter[], key: keyof User) {
+    const activeFilters = this.activeFilters$.value.set(
+      key,
+      selectedValues.map((it) => it.value)
+    );
+    this.activeFilters$.next(activeFilters);
+  }
+
+  private filterUsersBySearchTerm(user: User, searchTerm: string) {
```

Comment:
Rename this method - currently, it has a plural name although it only handles a single user.
Also `filter*` to me reads as if the method filters a list, which is not the case.
```suggestion
  private matchUserBySearchTerm(user: User, searchTerm: string) {
```